core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -53,6 +65,18 @@ class HadoopMapReduceCommitProtocol(jobId: String, path: String)
   @transient private var addedAbsPathFiles: mutable.Map[String, String] = null
 
   /**
+   * Tracks partitions with default path that have new files written into it by this task,
+   * e.g. a=1/b=2. Files under these partitions will be saved into staging directory and moved to
+   * destination directory at the end, if `runtimeOverwritePartition` is true.
+   */
+  @transient private var partitionPaths: mutable.Set[String] = null
```

Comment:
cc @ericl , `addedAbsPathFiles` only tracks partitions with custom path, we still need this `partitionPaths` to track partitions with default path